### PR TITLE
fluidsynth: update to 2.0.3

### PR DESCRIPTION
**Pull request by mojca**
#### Description

I was asked for an update via e-mail, but please don't merge until @RJVB reviews & approves the changes as that's basically his port and he's using it on Linux as well.

Also, there are some dependencies which I didn't check. This is a major version change, so it might break something. Some maintainers of dependent ports:
* @ryandesign: `lmms`
* @Ionic: `audacious-plugins`
* @kencu: `scummvm`
* @jmroot: `libsdl2_mixer` (has fluidsynth disabled though)
* @RJVB: `libVLC`

While the Portfile looks ok to me, I would really like the second opinion as I'm not even sure how to test the functionality.

* builds OK:
  * `audacious-plugins`
  * `gstreamer1-gst-plugins-bad`
  * `scummvm`
* broken:
  * `lmms` [ticket #58009](https://trac.macports.org/ticket/58009)
  * `denemo` (its dependency `qtplay` requires OS X 10.11 or older, [ticket #58010](https://trac.macports.org/ticket/58010))
* didn't try yet (too many deps):
  * `VLC`
We probably need to double-check dependencies as well. (One of them is reported missing and some might be used opportunistically?)

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.13.6 17G3025
Xcode 10.1 10B61

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

File: multimedia/fluidsynth/Portfile [hunk 1]
```diff
@@ -17,14 +17,13 @@ long_description    FluidSynth is a cross-platform real-time software synthesize
...
-use_bzip2           yes
+homepage            http://www.fluidsynth.org/
 
-checksums           rmd160  27b17e1e097004dd3cf8fb88d8e4499eeb339ea8 \
-                    sha256  d28b47dfbf7f8e426902ae7fa2981d821fbf84f41da9e1b85be933d2d748f601
+checksums           rmd160  dca66848a63d51ab6b26f5d649d7e8cd688514b6 \
+                    sha256  7fa7f07cb2a56480df2f80478d68f452a2748c7583aeff010e89af683075c708 \
+                    size    1373337
 
-depends_build       port:pkgconfig
```
Comment (RJVB) [inline multimedia/fluidsynth/Portfile]: I see that cmake looks for pkg-config; there is even an option to disable the use of pkg-config to find dependencies. In short: keep the port:pkgconfig build dependency please.

File: multimedia/fluidsynth/Portfile [hunk 2]
```diff
@@ -17,14 +17,13 @@ long_description    FluidSynth is a cross-platform real-time software synthesize
...
+depends_build       port:doxygen
```
Comment (RJVB) [inline multimedia/fluidsynth/Portfile]: doxygen is required to build the (API?) documentation, but I'm not seeing any indication that this is done in a regular build. Adding `-DCMAKE_DISABLE_FIND_PACKAGE_Doxygen=ON` to the configure arguments disables the doxygen lookup (and calling).

File: multimedia/fluidsynth/Portfile [hunk 3]
```diff
@@ -36,27 +35,22 @@ depends_lib         port:flac \
...
-patchfiles          yosemite-libtool.patch
-
 configure.args-append \
-                    --disable-dependency-tracking \
-                    --disable-pulse-support \
-                    --disable-jack-support \
-                    --disable-dbus-support
+                    -Denable-jack=OFF \
+                    -Denable-dbus=OFF \
+                    -Denable-pulseaudio=OFF
```
Comment (RJVB) [inline multimedia/fluidsynth/Portfile]: I'd append

```
                    -Denable-framework=OFF \
                    -DLIB_SUFFIX="" \
                    -DCMAKE_DISABLE_FIND_PACKAGE_Doxygen=ON
```

This disables the frameworks build (not appropriate for MacPorts IMHO), makes libraries go into `$prefix/lib` (and not $prefix/lib64) and disables the doxygen lookup.

**Comment by mojca** [comment]
In the meantime I found @RJVB's version at https://github.com/RJVB/macstrop/blob/master/multimedia/fluidsynth/Portfile
I guess I need to review the differences and act upon them.

File: multimedia/fluidsynth/Portfile (hunk 1 above)
Comment (mojca) [reply multimedia/fluidsynth/Portfile]: Sorry, that edit was totally stupid of me. I somehow connected this to "has something to do with autofoo magic". I'll revert.

File: multimedia/fluidsynth/Portfile (hunk 2 above)
Comment (mojca) [reply multimedia/fluidsynth/Portfile]: I've seen this in your version and what you do makes sense. I noticed that the port was opportunistically using the dependency, but I'm also way more happy to leave it out.

File: multimedia/fluidsynth/Portfile (hunk 3 above)
Comment (mojca) [reply multimedia/fluidsynth/Portfile]: OK.

**Comment by RJVB** [comment]
Guess what, I made my version while giving feedback on your proposal, figuring things as I went.

I did NOT test if my dependents (including VLC) work with the new version, I kind of assumed you did O:-)

**Comment by mojca** [comment]
I did not see your comments by the time I found your repository.

Btw: it seems that the URLs to fluidsynth soundfonts are broken. Any ideas about that?

**Comment by mojca** [comment]
@RJVB: this is unrelated to the PR in question, but it needs to be fixed:
```
--->  Verifying Portfile for VLC
Error: Unknown PortGroup: compress_workdir-1.0
--->  1 errors and 0 warnings found.
```

Homepage for `fluidsynth-soundfont` lists https://sourceforge.net/p/fluidsynth/wiki/SoundFont/ as the homepage, and that one is defunct.

**Comment by RJVB** [comment]
>@RJVB: this is unrelated to the PR in question, but it needs to be fixed:
>```
>--->  Verifying Portfile for VLC
>Error: Unknown PortGroup: compress_workdir-1.0
>--->  1 errors and 0 warnings found.

Yeah, but IMHO not in the Portfile: the test goes wrong somewhere else. The PortGroup is loaded only if it exists:

```
    if {[file exists ${filespath}/../../../_resources/port1.0/group/compress_workdir-1.0.tcl]} {
        PortGroup   compress_workdir 1.0
    }
```

That'valid Tcl so the validation procedure shouldn't be flagging errors.

**Comment by RJVB** [comment]
The fluid-soundfont homepage is now https://github.com/FluidSynth/fluidsynth/wiki/SoundFont .

**Comment by mojca** [comment]
@ryandesign: are you willing to look at `lmms`? It looks broken to me (but most likely for unrelated reasons, even though I didn't try building it against the old `fluidsynth` yet)

**Comment by RJVB** [comment]
FWIW, there's a 1.2.0.rc7 release for lmms according to livecheck ... but shouldn't you file an issue against that port on trac if it's the only one broken by the fluidsynth update?

(Would be good too to make it use Qt5.)

**Comment by mojca** [comment]
Yes, I should. One step at a time. It might not be the only port that's not working. I still didn't test two (VLC, denemo) due to the large number of dependencies. And I apparently forgot how to configure and use fluidsynth in the first place:

```
FluidSynth runtime version 2.0.2
Copyright (C) 2000-2018 Peter Hanappe and others.
Distributed under the LGPL license.
SoundFont(R) is a registered trademark of E-mu Systems, Inc.

fluidsynth: error: Unable to open file '/opt/local/share/soundfonts/default.sf2'
fluidsynth: error: Couldn't load soundfont file
fluidsynth: error: Failed to load SoundFont "/opt/local/share/soundfonts/default.sf2"
Type 'help' for help topics.

> fluidsynth: warning: No preset found on channel 0 [bank=0 prog=21]
fluidsynth: warning: No preset found on channel 0 [bank=0 prog=21]
fluidsynth: warning: No preset found on channel 1 [bank=0 prog=21]
fluidsynth: warning: No preset found on channel 1 [bank=0 prog=21]
```

**Comment by RJVB** [comment]
I have no problem with you merging this (or I cannot remember already 8-) )

Was there a reason for another revbump or was that something you forgot for the upgrade to 2.0.2?

**Comment by RJVB** [comment]
FWIW: https://github.com/FluidSynth/fluidsynth/issues/476#issuecomment-459707336

**Pull request merged**